### PR TITLE
Implement ability to customize contest problem labels; #1124

### LINKED ISSUE #1124: Custom contest problem labels
Right now we hardcode "1", "2", "3", etc. Some contests (like ICPC) may wish to use different labels, like "A", "B", or "C". I'm in favour of allowing user to override by a `label` field in `ContestProblem`, and give the contest format the choice when that is null by some sort of `get_label_for_problem(n : int) -> str` function.

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Closes #1124.

### CODE REVIEW

File: judge/models/contest.py [hunk 1]
```diff
@@ -113,6 +115,10 @@ class Contest(models.Model):
                               help_text=_('A JSON object to serve as the configuration for the chosen contest format '
                                           'module. Leave empty to use None. Exact format depends on the contest format '
                                           'selected.'))
+    problem_label_script = models.TextField(verbose_name='contest problem label script', blank=True,
+                                            help_text='A custom LUA function to generate problem labels. Requires a '
+                                                      'single function with an integer paramater, the contest problem '
```
Comment (Xyene) [inline judge/models/contest.py]: typo in `parameter`

File: judge/models/contest.py [hunk 2]
```diff
@@ -113,6 +115,10 @@ class Contest(models.Model):
                               help_text=_('A JSON object to serve as the configuration for the chosen contest format '
                                           'module. Leave empty to use None. Exact format depends on the contest format '
                                           'selected.'))
+    problem_label_script = models.TextField(verbose_name='contest problem label script', blank=True,
+                                            help_text='A custom LUA function to generate problem labels. Requires a '
```
Comment (Xyene) [inline judge/models/contest.py]: `Lua`

File: judge/models/contest.py [hunk 3]
```diff
@@ -113,6 +115,10 @@ class Contest(models.Model):
...
+                                                      'single function with an integer paramater, the contest problem '
+                                                      '"order", and returns a string, the label.')
```
Comment (Xyene) [inline judge/models/contest.py]: 0-indexed or 1-indexed?

File: judge/models/contest.py [hunk 4]
```diff
@@ -122,12 +128,29 @@ def format_class(self):
     def format(self):
         return self.format_class(self, self.format_config)
 
+    @cached_property
+    def get_label_for_problem(self):
+        def DENY_ALL(obj, attr_name, is_setting):
+            raise AttributeError()
+        lua = LuaRuntime(attribute_filter=DENY_ALL, register_eval=False, register_builtins=False)
+        return lua.eval(self.problem_label_script or settings.DMOJ_CONTEST_DEFAULT_PROBLEM_LABEL_SCRIPT)
```
Comment (Xyene) [inline judge/models/contest.py]: I think this should be queried from the contest format. All ICPC contests should have letters, for instance.

File: judge/models/contest.py (hunk 3 above)
Comment (Ninjaclasher) [reply judge/models/contest.py]: This is dependant on the contest setter (if labels are generated with `problem.order`).

**Pull request merged**